tests/datadriven/__init__.py

```diff
@@ -86,6 +93,130 @@ def assertEqual(self, a, b):
+
+    def assertIs(self, a, b):
+        """
+        Tests that a is b.  If it is not, raise an assertion error.
+        """
+        if a is not b:
+            raise AssertionError("{} is not {}".format(a, b))
+
+    def assertIsNot(self, a, b):
+        """
+        Tests that a is not b.  If a is b, raise an assertion error.
+        """
+        if a is b:
+            raise AssertionError("{} is {}".format(a, b))
+
+    def assertIsNone(self, x):
+        """
+        Tests that x is None.  If x is not, raise an assertion error.
+        """
+        if x is not None:
+            raise AssertionError("{} is not None".format(x))
+
+    def assertIsNotNone(self, x):
+        """
+        Tests that x is not None.  If x is None, raise an assertion error.
+        """
+        if x is None:
+            raise AssertionError("{} is None".format(x))
+
+    def assertIn(self, a, b):
+        """
+        Tests that a is in b.  If a is not in b, raise an assertion error.
+        """
+        if a not in b:
+            raise AssertionError("{} not in {}".format(a, b))
+
+    def assertNotIn(self, a, b):
+        """
+        Tests that a is not in b.  If a is in b, raise an assertion error.
+        """
+        if a in b:
+            raise AssertionError("{} in {}".format(a, b))
+
+    def assertIsInstance(self, a, b):
+        """
+        Tests that a is an instance of b.  If a is not an instance of b,
+        raise an assertion error.
+        """
+        if not isinstance(a, b):
+            raise AssertionError("{} is not instance of {}".format(a, b))
+
+    def assertNotIsInstance(self, a, b):
+        """
+        Tests that a is not an instance of b.  If a is an instance of b,
+        raise an assertion error.
+        """
+        if isinstance(a, b):
+            raise AssertionError("{} is instance of {}".format(a, b))
+
+    def assertGreater(self, a, b):
+        """
+        Tests that a is greater than b.  If not, raise an assertion error.
+        """
+        if a <= b:
+            raise AssertionError("{} is not greater than {}".format(a, b))
+
+    def assertGreaterEqual(self, a, b):
+        """
+        Tests that a is greater than or equal to b.
+        If not, raise an assertion error.
+        """
+        if a < b:
+            raise AssertionError(
+                "{} is not greater than or equal to {}".format(a, b))
+
+    def assertLess(self, a, b):
+        """
+        Tests that a is less than b.  If not, raise an assertion error.
+        """
+        if a >= b:
+            raise AssertionError("{} is not less than {}".format(a, b))
+
+    def assertLessEqual(self, a, b):
+        """
+        Tests that a is less than or equal to b.
+        If not, raise an assertion error.
+        """
+        if a > b:
+            raise AssertionError(
+                "{} is not less than or equal to {}".format(a, b))
+
+    @contextlib.contextmanager
+    def assertRaises(self, exceptionType):
```

Comment:
This is the syntax to allow us to use the `with` keyword when using `assertRaises`